### DISCUSSION TITLE
How to use js.liquid files?

### DISCUSSION THREAD

**Discussion by explorier** [krjo/shopify-webpack-dev-workflow#13] [Q&A]
This repo looks great so far! Trying to migrate a legacy theme from Slate to this setup. However, the original code uses a theme.js.liquid file as the main entry point, and there's some liquid code in there I'll need to keep. Do you have a recommended way to handle this? Wondering if I should modify the Webpack config. Thanks!

**Comment by explorier** [comment]
Ah, I may have misstated that a bit - despite having worked on this codebase for a bit the old Slate/Shopify way of handling JS etc is still confusing me! The old theme didn't actually compile the JS, just had a Gulpfile to take in some SCSS. At any rate, I'm realizing the best approach may be to abandon the js.liquid files and throw that data into the actual liquid files for JS to access from there. 🤞🏼

**Reply by krjo** [reply]
I agree. I didn't bother including a liquid compiler. I would recommend adding a shop-api.liquid snippet or just adding a global object in a script tag in the <head> in theme.liquid. That shop-api file can hold an object of variables with values that are output by liquid. That way you can use liquid data in your js file. Also Shopify is deprecating the. For example:

`{% render "shop-api" %}`

In that file:

`<script>
window.shopApi = {
template: '{{ template }}',
settings: {
someThemeSettingCheckbox: {{ settings.someThemeSetting | json }},
{% if request.page_type == 'product' %}
product: {
productType: '{{ product.type }}'
}
{% endif %}
}
}
</script>`

**Reply by krjo** [reply]
Also, if you want use liquid / theme settings to assign to values to some css variables, you can just use CSS variables, assigned to :root, which you can also put in theme.liquid. Like this: 

`<style>
:root {
  --body-font-color: {{ settings.body_font_color }};
}
</style>`

Used like: 

`
p {
color: var(--body-font-color);
}
`
